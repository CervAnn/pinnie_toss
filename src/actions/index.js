export const addTeams = teams => ({
  type: 'ADD_TEAMS',
  teams
})

export const pickRandom = team => ({
  type: 'PICK_RANDOM',
  team
})