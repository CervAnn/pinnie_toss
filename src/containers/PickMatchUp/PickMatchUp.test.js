import React from "react";
import { shallow } from "enzyme";
import {
  PickMatchUp,
  mapStateToProps,
  mapDispatchToProps
} from "./PickMatchUp";
import { addSeason, addUnderDogs, addWinners } from "../../actions/index";

describe("PickMatchUp", () => {
  let wrapper, mockAddSeason, mockAddUnderDogs, mockAddWinners;

  beforeEach(() => {
    mockAddSeason = jest.fn();
    mockAddUnderDogs = jest.fn();
    mockAddWinners = jest.fn();
    wrapper = shallow(
      <PickMatchUp
        addSeason={mockAddSeason}
        addUnderDogs={mockAddUnderDogs}
        addWinners={mockAddWinners}
        teams={[]}
        league={{}}
        season={2019}
        underDogs={[]}
        winners={[]}
      />
    );
  });

  it("should match the snapshot with all data passed in correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call filterUnderDogs on a button click", () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().filterUnderDogs = jest.fn();
    wrapper
      .find("button")
      .at(1)
      .simulate("click", mockEvent);

    expect(wrapper.instance().filterUnderDogs).toHaveBeenCalled();
  });

  it("should call filterUnderDogs on an input click", () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().filterUnderDogs = jest.fn();
    wrapper
      .find("input")
      .at(1)
      .simulate("click", mockEvent);

    expect(wrapper.instance().filterUnderDogs).toHaveBeenCalled();
  });

  it("should call filterWinningestTeams on a button click", () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().filterWinningestTeams = jest.fn();
    wrapper
      .find("button")
      .at(1)
      .simulate("click", mockEvent);

    expect(wrapper.instance().filterWinningestTeams).toHaveBeenCalled();
  });

  it("should call filterWinningestTeams on an input click", () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().filterWinningestTeams = jest.fn();
    wrapper
      .find("input")
      .at(1)
      .simulate("click", mockEvent);

    expect(wrapper.instance().filterWinningestTeams).toHaveBeenCalled();
  });

  describe("mapStateToProps", () => {
    it("should return an object with the contents of the store", () => {
      let mockStore = {
        teams: [
          {
            idTeam: "134920"
          },
          {
            idTeam: "134922"
          }
        ],
        league: {
          strLeague: "NFL"
        },
        season: 2019,
        underDogs: [],
        winners: [
          {
            idTeam: "134920",
            strLeague: "NFL",
            strTeam: "New England Patriots",
            strTeamBadge:
              "https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
          },
          {
            idTeam: "134922",
            strLeague: "NFL",
            strTeam: "Baltimore Ravens",
            strTeamBadge:
              "https://www.thesportsdb.com/images/media/team/badge/einz3p1546172463.png"
          }
        ]
      };

      expect(mapStateToProps(mockStore)).toEqual(mockStore);
    });
  });

  describe("mapDispatchToProps", () => {
    it("should dispatch addSeason when called", () => {
      let mockSeason = 2019;

      let mockDispatch = jest.fn();
      let actionToDispatch = addSeason(mockSeason);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addSeason(2019);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it("should dispatch addUnderDogs when called", () => {
      let mockTeams = [
        {
          idTeam: "134920",
          strLeague: "NFL",
          strTeam: "New England Patriots",
          strTeamBadge:
            "https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
        },
        {
          idTeam: "134922",
          strLeague: "NFL",
          strTeam: "Baltimore Ravens",
          strTeamBadge:
            "https://www.thesportsdb.com/images/media/team/badge/einz3p1546172463.png"
        }
      ];

      let mockDispatch = jest.fn();
      let actionToDispatch = addUnderDogs(mockTeams);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addUnderDogs(mockTeams);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it("should dispatch addWinners when called", () => {
      let mockTeams = [
        {
          idTeam: "134920",
          strLeague: "NFL",
          strTeam: "New England Patriots",
          strTeamBadge:
            "https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
        },
        {
          idTeam: "134922",
          strLeague: "NFL",
          strTeam: "Baltimore Ravens",
          strTeamBadge:
            "https://www.thesportsdb.com/images/media/team/badge/einz3p1546172463.png"
        }
      ];

      let mockDispatch = jest.fn();
      let actionToDispatch = addWinners(mockTeams);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addWinners(mockTeams);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
