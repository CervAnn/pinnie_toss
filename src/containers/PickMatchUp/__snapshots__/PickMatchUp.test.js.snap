// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickMatchUp should match the snapshot with all data passed in correctly 1`] = `
<section
  className="PickMatchUp"
>
  <h1
    id="pickMatchUp-text"
  >
    Do you like an underdog?
  </h1>
  <div
    id="pickUnderDog-buttons-container"
  >
    <Link
      id="yourTeam-link"
      to="/yourTeam"
    >
      <input
        id="gloves"
        onClick={[Function]}
        type="submit"
        value=""
      />
      <button
        id="gloves-button"
        onClick={[Function]}
      >
        Absolutely!
      </button>
    </Link>
  </div>
  <div
    id="pickSafeBet-buttons-container"
  >
    <Link
      id="yourTeam-link"
      to="/yourTeam"
    >
      <input
        id="trophy"
        onClick={[Function]}
        type="submit"
        value=""
      />
      <button
        id="trophy-button"
        onClick={[Function]}
      >
        No, I Prefer A Reliable Win
      </button>
    </Link>
  </div>
</section>
`;
